woocommerce/woocommerce-android
# Upgrade gradle to 7.1.1 & Android Gradle Plugin to 4.2.2

This PR upgrades Gradle to 7.1.1 and Android Gradle Plugin to 4.2.2. The changes are minimal and mostly generated by the Gradle upgrade script `./gradlew wrapper --gradle-version=7.1.1 --distribution-type=all`. As part of this PR `buildToolsVersion` is removed since it now uses the version specified by the Android Gradle plugin.

**To test:**
* Smoke test the app
* Technically we could smoke test the app with fluxc composite build as well using the [Gradle 7 branch of FluxC](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2084), however I imagine these branches are out of sync now and FluxC branch will be tested as part of [WordPress-Android](https://github.com/wordpress-mobile/WordPress-Android/pull/15076), so we should be covered. [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2084) does not have any implementation changes, so we are just trying to make sure that composite build still works.

**Tasks:**
- [x] Since we are upgrading Sentry plugin, we should also upgrade Sentry to the latest version to avoid any potential issues with the version mismatch
- [x] Verify debug symbols are uploaded correctly to Sentry
- [x] A lot has changed in AGP 4.2, we'll want to ensure none of those changes will create a regression. I am still going over release notes and checking each repository for issues and so far I haven't found any issues in WCAndroid. However, we are still far from calling this a good update yet.
- [x] Remove the explicit R8 version after confirming that the release build works fine without it. (more info [here](https://github.com/woocommerce/woocommerce-android/pull/4173))

**Update release notes:**
